### PR TITLE
feat: Oauth with PKCE

### PR DESCRIPTION
**Pull request by AdityaPimpalkar**
ref - #4437

I've commented out code which will be needed once #4594 is implemented. The database calls depends on it.

### CODE REVIEW

File: packages/twenty-server/src/engine/core-modules/auth/dto/exchange-auth-code.input.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { ArgsType, Field } from '@nestjs/graphql';
+
+@ArgsType()
+export class ExchangeAuthCodeInput {
+  @Field(() => String)
+  authorizationCode: string;
+
```
Comment (FelixMalfait) [inline packages/twenty-server/src/engine/core-modules/auth/dto/exchange-auth-code.input.ts:7]: Could you also support non PKCE / traditional auth as part of this or maybe do it in another PR but at least in this one introduce the auth type (I'm not sure where that lives? On the app there would be the corresponding auth type maybe?  codeVerifier being only for PKCE

File: packages/twenty-server/src/engine/core-modules/auth/services/auth.service.ts [hunk 2]
```diff
@@ -173,6 +179,40 @@ export class AuthService {
...
+
+    const client = apps.find((app) => app.id === clientId);
+
+    if (!client) {
+      throw new NotFoundException(`Invalid client '${clientId}'`);
+    }
+
+    const authorizationCode = crypto.randomBytes(42).toString('hex');
+
+    const expiresAt = addMilliseconds(new Date().getTime(), ms('1m'));
```
Comment (FelixMalfait) [inline packages/twenty-server/src/engine/core-modules/auth/services/auth.service.ts:207]: Seems a bit short ;-)

**Review by FelixMalfait — COMMENTED** [review]
Couldn't test but looks like a great start! If you're sure this doesn't break anything we can merge step-by-step, let me know

**Comment by FelixMalfait** [comment]
(if you do want to merge after fixes let's at least fix linter)

File: packages/twenty-server/src/engine/core-modules/auth/services/auth.service.ts (hunk 2 above)
Comment (AdityaPimpalkar) [reply packages/twenty-server/src/engine/core-modules/auth/services/auth.service.ts]: 5m?

File: packages/twenty-server/src/engine/core-modules/auth/dto/exchange-auth-code.input.ts (hunk 1 above)
Comment (AdityaPimpalkar) [reply packages/twenty-server/src/engine/core-modules/auth/dto/exchange-auth-code.input.ts:7]: Best to do it separate PR?

**Review by FelixMalfait — APPROVED** [review]
